File: leetcode/is-palindrome.go
package leetcode

func isPalindrome(s string) bool {
	left, right := 0, len(s)-1
	isAlphaNumeric := func(c byte) bool {
		return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')
	}
	for left < right {
		if !isAlphaNumeric(s[left]) {
			left++
			continue
		} else if !isAlphaNumeric(s[right]) {
			right--
			continue
		}
		if (s[left] | 32) != (s[right] | 32) {
			return false
		}
		left++
		right--
	}
	return true
}

File: leetcode/is-palindrome.ts
function isPalindrome(s: string): boolean {
  let left = 0;
  let right = s.length - 1;

  const isAlphaNumeric = (char: string): boolean => /[a-zA-Z0-9]/.test(char);
  while (left < right) {
    if (!isAlphaNumeric(s[left])) {
      left++;
      continue;
    } else if (!isAlphaNumeric(s[right])) {
      right--;
      continue;
    }

    if (s[left].toLocaleLowerCase() !== s[right].toLocaleLowerCase()) {
      return false;
    }
    left++;
    right--;
  }
  return true;
}
